content/blog/exposing-an-application-using-ingress-and-tls-termination-on-kubernetes-in-hpe-greenlake-for-private-cloud-enterprise.md

```diff
@@ -0,0 +1,528 @@
+
+### Set up the load balancer with MetalLB
+
+You can install MetalLB and set up the load balancer in the K8s cluster by following up the blog post [Setting up the load balancer with MetalLB](https://developer.hpe.com/blog/set-up-load-balancer-with-metallb-in-hpe-greenlake-for-private-cloud-enterprise/).
+
+H﻿ere is the deployed MetalLB to the namespace *metallb-system* in the cluster:
+
+```shell
+$ kubectl get all -n metallb-system
+NAME                              READY   STATUS    RESTARTS   AGE
+pod/controller-57b4fdc957-56wv8   1/1     Running   0          22m
+pod/speaker-c7sgk                 1/1     Running   0          22m
+pod/speaker-dtlpm                 1/1     Running   0          22m
+pod/speaker-gxccz                 1/1     Running   0          22m
+pod/speaker-pwl87                 1/1     Running   0          22m
+pod/speaker-rvvkz                 1/1     Running   0          22m
+pod/speaker-wxd5n                 1/1     Running   0          22m
+
+NAME                      TYPE        CLUSTER-IP     EXTERNAL-IP   PORT(S)   AGE
+service/webhook-service   ClusterIP   10.102.54.20   <none>        443/TCP   22m
+
+NAME                     DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR            AGE
+daemonset.apps/speaker   6         6         6       6            6           kubernetes.io/os=linux   22m
+
+NAME                         READY   UP-TO-DATE   AVAILABLE   AGE
+deployment.apps/controller   1/1     1            1           22m
+
+NAME                                    DESIRED   CURRENT   READY   AGE
+replicaset.apps/controller-57b4fdc957   1         1         1       22m
+```
+
+You can see the range of virtual IP addresses, *"10.6.115.251-10.6.115.254"*, defined in the CRD resource *IPAddressPool*, and the layer 2 service IP address announcement in the CRD resource *L2Advertisement*:
+
+```shell
+$ kubectl get ipaddresspools -n metallb-system
+NAME       AUTO ASSIGN   AVOID BUGGY IPS   ADDRESSES
+cfe-pool   true          false             ["10.6.115.251-10.6.115.254"]
+
+$ kubectl get l2advertisements -n metallb-system
+NAME           IPADDRESSPOOLS   IPADDRESSPOOL SELECTORS   INTERFACES
+cfe-l2advert   ["cfe-pool"]
+```
+
+### Deploy Nginx Ingress controller
+
+In order for an Ingress to work in the cluster, there must be an Ingress controller being deployed and running. It's the Ingress controller that accesses the certificate and the routing rules defined on the Ingress resource and makes them part of its configuration. 
+
+A variety of Ingress controllers are available for deployment in the cluster, including [Traefik](https://doc.traefik.io/traefik/providers/kubernetes-ingress/), [HAProxy](https://github.com/haproxytech/kubernetes-ingress#readme) and [Nginx Ingress controller](https://www.nginx.com/products/nginx-ingress-controller/). Execute the command below to install the Nginx Ingress controller to the cluster using helm:
```

Comment:
Is "helm" a product name? If so, it should have an initial capital letter, i.e. Helm.